libraries/AP_HAL/CANIface.cpp

```diff
@@ -99,6 +99,8 @@ AP_HAL::CANFrame::CANFrame(uint32_t can_id, const uint8_t* can_data, uint8_t dat
         canfd(canfd_frame)
 {
     if ((can_data == nullptr) || (data_len == 0) || (data_len > MaxDataLen)) {
+        dlc = 0;
```

Comment:
Is this a "programmer has screwed up" situation?

Reply:
Humans.. pffft

Reply:
This PR is now on top of PR https://github.com/ArduPilot/ardupilot/pull/23328